lib/task-graph.js

```diff
@@ -360,7 +377,11 @@ function taskGraphFactory(
             assert.arrayOfObject(self.definition.tasks, 'Graph.tasks');
             return Promise.map(self.definition.tasks, function(taskData) {
                 if (!_.has(taskData, 'taskDefinition')) {
-                    return store.getTaskDefinition(taskData.taskName)
+                    var taskName = taskData.taskName;
+                    if(taskName){
+                        taskName = getTaskName(taskName);
```

Comment:
taskName in this function have two meanings.  former taskName is taskName input, latter is true taskName, it's not so good for clean code. so you could naming former one as taskNameInput, the latter is taskName. Otherwise, it's a little werid that get 'taskName' from 'taskName'